driver-core/src/main/com/mongodb/client/model/expressions/DocumentExpression.java

```diff
@@ -25,86 +25,487 @@
+     * or if the document {@linkplain #has} no such field.
+     *
+     * @mongodb.server.release 5.0
+     * @param fieldName the name of the field.
+     * @param other the other value.
+     * @return the resulting value.
+     */
     default DateExpression getDate(final String fieldName, final Instant other) {
         return getDate(fieldName, of(other));
     }
 
+    /**
+     * Returns the {@linkplain DocumentExpression document} value of the field
+     * with the provided {@code fieldName}.
+     *
+     * <p>Warning: The type and presence of the resulting value is not
+     * enforced by the API. The use of this method is an
+     * unchecked assertion that the field is present and
+     * the field value is of the specified type.
+     *
+     * @mongodb.server.release 5.0
+     * @param fieldName the name of the field.
+     * @return the resulting value.
+     */
+    @MqlUnchecked({PRESENT, TYPE})
     DocumentExpression getDocument(String fieldName);
+
+    /**
+     * Returns the {@linkplain DocumentExpression document} value of the field
+     * with the provided {@code fieldName},
+     * or the {@code other} value if the field is not a (child) document
+     * or if the document {@linkplain #has} no such field.
+     *
+     * <p>Note: Any field considered to be a document by this API
+     * will also be considered a map, and vice-versa.
+     *
+     * @mongodb.server.release 5.0
+     * @param fieldName the name of the field.
+     * @param other the other value.
+     * @return the resulting value.
+     */
     DocumentExpression getDocument(String fieldName, DocumentExpression other);
 
+    /**
+     * Returns the {@linkplain DocumentExpression document} value of the field
+     * with the provided {@code fieldName},
+     * or the {@code other} value if the field is not a (child) document
+     * or if the document {@linkplain #has} no such field.
+     *
+     * <p>Note: Any field considered to be a document by this API
+     * will also be considered a map, and vice-versa.
+     *
+     * @mongodb.server.release 5.0
+     * @param fieldName the name of the field.
+     * @param other the other value.
+     * @return the resulting value.
+     */
     default DocumentExpression getDocument(final String fieldName, final Bson other) {
         return getDocument(fieldName, of(other));
     }
 
+    /**
+     * Returns the {@linkplain MapExpression map} value of the field
+     * with the provided {@code fieldName}.
+     *
+     * <p>Warning: The type and presence of the resulting value is not
+     * enforced by the API. The use of this method is an
+     * unchecked assertion that the field is present and
+     * the field value is of the specified type.
+     *
+     * @mongodb.server.release 5.0
+     * @param fieldName the name of the field.
+     * @return the resulting value.
+     * @param <T> the type.
+     */
+    @MqlUnchecked({PRESENT, TYPE, TYPE_ARGUMENT})
```

Comment:
It seems useful to remind users to carefully look at the type argument as well, as we do elsewhere.

Reply:
If it deemed important to talk about the raw type and the parameterization instead of simply talking about the type, then shouldn't the same be done in the javadoc comments? The current wording is _"The use of this method is an unchecked assertion that ... the field value is of the specified type"_ (this example is from `DocumentExpression.getMap(String)`).